legendary_eureka.py

```diff
@@ -34,6 +39,99 @@ class Glassdoor(Resource):
     def __init__(self):
         Resource.__init__(self, 'glassdoor')
 
-    # TODO
-    def get_jobs(self):
-        pass
+    # TODO: wrap requests.post parts with try / except.
+    def get_jobs(self, job_title: str, job_location: str = None, job_type: str = ''):
+        """
+        using part of code from: https://gist.github.com/scrapehero/352286d0f9dee87990cd45c3f979e7cb
+        :param job_title: for example: 'python developer'
+        :param job_type: optional, 'fulltime' or ___ or ___ TODO: get job_type possibilities
+        :param job_location: Jerusalem (Israel), Tel Aviv-Yafo (Israel), Haifa (Israel), Ashdod (Israel),
+         Bnei Brak (Israel), Netanya (Israel), Bene Beraq (Israel), Bat Yam (Israel), Ramat Gan(Israel),
+         Ashqelon (Israel).
+        """
+
+        headers = {'accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8',
+                   'accept-encoding': 'gzip, deflate, sdch, br',
+                   'accept-language': 'en-GB,en-US;q=0.8,en;q=0.6',
+                   'referer': 'https://www.glassdoor.com/',
+                   'upgrade-insecure-requests': '1',
+                   'user-agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko)'
+                                 ' Ubuntu Chromium/51.0.2704.79 Chrome/51.0.2704.79 Safari/537.36',
+                   'Cache-Control': 'no-cache',
+                   'Connection': 'keep-alive'
+                   }
+
+        location_headers = {
+            'accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.01',
+            'accept-encoding': 'gzip, deflate, sdch, br',
+            'accept-language': 'en-GB,en-US;q=0.8,en;q=0.6',
+            'referer': 'https://www.glassdoor.com/',
+            'upgrade-insecure-requests': '1',
+            'user-agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko)'
+                          ' Ubuntu Chromium/51.0.2704.79 Chrome/51.0.2704.79 Safari/537.36',
+            'Cache-Control': 'no-cache',
+            'Connection': 'keep-alive'
+        }
+        data = {"term": 'Israel',
+                "maxLocationsToReturn": 10}
+
+        location_url = "https://www.glassdoor.co.in/findPopularLocationAjax.htm?"
+
+        # Getting location id for search location
+        print("Fetching location details")
+        location_response = requests.post(location_url, headers=location_headers, data=data).json()
+
+        if job_location in [dictionary['label'] for dictionary in location_response]:
+            location_response = [dictionary for dictionary in location_response
+                                 if job_location in dictionary['label']]
+        else:
+            print('Job location not found, printing jobs from all locations in Israel..')
```

Comment:
We want to get all the jobs in Israel by default, filtering by a location is an extra step

Reply:
Found a way to do that, searched a little deeper in the post request .